specs/sharding/beacon-chain.md

```diff
@@ -103,786 +87,558 @@ The following values are (non-configurable) constants used throughout the specif
+| `ROOT_OF_UNITY` | `pow(PRIMITIVE_ROOT_OF_UNITY, (MODULUS - 1) // int(SAMPLES_PER_BLOB * FIELD_ELEMENTS_PER_SAMPLE), MODULUS)` |
 
 ## Configuration
 
 Note: Some preset variables may become run-time configurable for testnets, but default to a preset while the spec is unstable.  
-E.g. `INITIAL_ACTIVE_SHARDS`, `MAX_SAMPLES_PER_BLOB` and `TARGET_SAMPLES_PER_BLOB`.
+E.g. `ACTIVE_SHARDS` and `SAMPLES_PER_BLOB`.
 
-## Updated containers
+### Time parameters
 
-The following containers have updated definitions to support Sharding.
+| Name | Value | Unit | Duration |
+| - | - | :-: | :-: |
+| `SECONDS_PER_SLOT` | `uint64(8)` | seconds | 8 seconds |
 
-### `AttestationData`
-
-```python
-class AttestationData(Container):
-    slot: Slot
-    index: CommitteeIndex
-    # LMD GHOST vote
-    beacon_block_root: Root
-    # FFG vote
-    source: Checkpoint
-    target: Checkpoint
-    # Hash-tree-root of ShardBlob
-    shard_blob_root: Root  # [New in Sharding]
-```
 
-### `BeaconBlockBody`
+## Containers
 
-```python
-class BeaconBlockBody(bellatrix.BeaconBlockBody):  # [extends Bellatrix block body]
-    shard_proposer_slashings: List[ShardProposerSlashing, MAX_SHARD_PROPOSER_SLASHINGS]
-    shard_headers: List[SignedShardBlobHeader, MAX_SHARDS * MAX_SHARD_HEADERS_PER_SHARD]
-```
+### New Containers
 
-### `BeaconState`
+#### `IntermediateBlockBid`
 
 ```python
-class BeaconState(bellatrix.BeaconState):
-    # Blob builder registry.
-    blob_builders: List[Builder, BLOB_BUILDER_REGISTRY_LIMIT]
-    blob_builder_balances: List[Gwei, BLOB_BUILDER_REGISTRY_LIMIT]
-    # A ring buffer of the latest slots, with information per active shard.
-    shard_buffer: Vector[List[ShardWork, MAX_SHARDS], SHARD_STATE_MEMORY_SLOTS]
-    shard_sample_price: uint64
-```
+class IntermediateBlockBid(Container):
+    slot: Slot
+    parent_block_root: Root
 
-## New containers
+    execution_payload_root: Root
 
-### `Builder`
+    sharded_data_commitment_root: Root # Root of the sharded data (only data, not beacon/intermediate block commitments)
 
-```python
-class Builder(Container):
-    pubkey: BLSPubkey
-    # TODO: fields for either an expiry mechanism (refunding execution account with remaining balance) 
-    #  and/or a builder-transaction mechanism.
-```
+    sharded_data_commitment_count: uint64 # Count of sharded data commitments
 
-### `DataCommitment`
+    bid: Gwei # Block builder bid paid to proposer
 
-```python
-class DataCommitment(Container):
-    # KZG10 commitment to the data
-    point: BLSCommitment
-    # Length of the data in samples
-    samples_count: uint64
+    validator_index: ValidatorIndex # Validator index for this bid
+    
+    # Block builders use an Eth1 address -- need signature as
+    # block builder fees and data gas base fees will be charged to this address
+	signature_y_parity: bool
+	signature_r: uint256
+	signature_s: uint256    
 ```
 
-### `AttestedDataCommitment`
+#### `IntermediateBlockBidWithRecipientAddress`
```

Comment:
What do we actually gain from this being a separate class? Why not just stick the ethereum address into the bid itself?

Reply:
It simplifies processing. We need to iterate through lists that contain the commitment and the recipient address as we process the payments. But yes it's not a huge difference either way.